src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -561,6 +564,12 @@ private void MessageRaised(object sender, BuildMessageEventArgs e)
                     project.OutputPath = outputPath;
                 }
             }
+
+            // Detect markers that require special attention from a customer.
+            if (_immediateMessageMarkers.Any(marker => message.IndexOf(marker, StringComparison.Ordinal) > 0))
```

Comment:
I _think_ if you can convert the string to a span/memory, then `Memory.IndexOfAny(StringValues)` will be much more performant here and on 598 below.

Reply:
I think a regex might actually be the best path forward here (😱). I don't know another way to  do an equivalent of `IndexOfAny(string[])`. We can use the .NET 8 source-generated ones though!

Reply:
so should I switch to regex here?

Reply:
I think so. Since it's scanning lots of messages I think it's worth pursuing the perf improvements we'd get there.